components/OtherUserSpectrum.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+  <li>
+    <slot></slot>
+    <span class="other-user-spectrum-gradient" :class="gradientStyle">
+      <span class="other-user-circle" :style="{ left: positionStyle }"></span>
+    </span>
+  </li>
+</template>
+
+<script>
+export default {
+  props: {
+    /* all gradientStyle classes passed to this component are generated in ~/assets/scss/gradients.scss */
+    gradientStyle: {
+      type: String,
+      required: true,
+      default: null,
+    },
+    otherUserPosition: {
+      type: Number,
+      required: true,
+      default: null,
+    },
+  },
+  computed: {
+    positionStyle() {
+      let n = Math.round(this.otherUserPosition * 100)
+      n = Math.max(10, Math.min(n, 90))
```

Comment:
I don't follow this 10 < n < 90 clamping calculation... it seems like we are loosing accuracy this way. why?

Reply:
Because the white ball gets rendered outside of the spectrum if the value is 0 or 100. To maintain the accuracy we can use the Processing mapping function to remap the actual value to the 10–90 range

Reply:
Ok, I'd prefer mapping over clamping. This is a nice to have, not high priority